### PR TITLE
[bitnami/external-dns] Add hetzner as provider

### PR DESCRIPTION
**Pull request by morremeyer**
**Description of the change**

This PR adds the possibility to configure [Hetzner DNS](https://dns.hetzner.com/) as provider.

**Benefits**

Users can now simply use Hetzner DNS together with external-dns.

**Possible drawbacks**

None.

**Applicable issues**

None.

**Additional information**

I wrote the configuration so that using a pre-defined secret will be preferred if defined.

**Question**: I didn’t find any information about the `values-production.yaml` file except that I should update it, too. Can someone tell me where it is used?

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [x] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

### CODE REVIEW

File: bitnami/external-dns/README.md [hunk 1]
```diff
@@ -118,6 +118,9 @@ The following table lists the configurable parameters of the external-dns chart
...
+| `hetzner.secretName`                   | When using the Hetzner provider, specify the existing secret which contains your token. Disables the usage of `hetzner.token` (optional)                                                                        | `""`                                                    |
+| `hetzner.secretKey`                    | When using the Hetzner provider with an existing secret, specify the key name (optional)                                                                                                                        | `"hetzner_token"`                                       |
+| `hetzner.token`                        | When using the Hetzner provider, specify your token here. (required when `hetzner.secretName` is not provided. In this case a new secret will be created holding the token.)                                    | `""`                                                    |
```
Comment (miguelaeh) [inline bitnami/external-dns/README.md:123]: Please format the table properly

File: bitnami/external-dns/values.yaml [hunk 2]
```diff
@@ -41,7 +41,7 @@ sources:
   # - contour-httpproxy
 
 ## DNS provider where the DNS records will be created. Available providers are:
-## - alibabacloud, aws, azure, azure-private-dns, cloudflare, coredns, designate, digitalocean, google, infoblox, rfc2136, transip
+## - alibabacloud, aws, azure, azure-private-dns, cloudflare, coredns, designate, digitalocean, google, hetzner, infoblox, rfc2136, transip
```
Comment (miguelaeh) [inline bitnami/external-dns/values.yaml:44]: I think this change should be done in the values-production too.

File: bitnami/external-dns/templates/_helpers.tpl [hunk 3]
```diff
@@ -130,6 +130,8 @@ Return true if a secret object should be created
     {{- true -}}
```
Comment (miguelaeh) [inline bitnami/external-dns/templates/_helpers.tpl:130]: In this file you must also modify the ` "external-dns.secretName"`

File: bitnami/external-dns/templates/secret.yaml [hunk 4]
```diff
@@ -25,6 +25,9 @@ data:
   {{- if eq .Values.provider "google" }}
   credentials.json: {{ .Values.google.serviceAccountKey | b64enc | quote }}
   {{- end }}
+  {{- if and (eq .Values.provider "hetzner") (not .Values.hetzner.secretName) }}
+  hetzner_token: {{ .Values.hetzner.token | b64enc | quote }}
```
Comment (miguelaeh) [inline bitnami/external-dns/templates/secret.yaml:29]: Could you please add validation to check the token is provided like in the rest of the cases? Check the _helpers.tpl

**Review by miguelaeh — CHANGES_REQUESTED** [review]
Hi,
Thank you very much for the PR!!
Please take a look at my comments.

File: bitnami/external-dns/README.md (hunk 1 above)
Comment (morremeyer) [reply bitnami/external-dns/README.md:123]: I don’t know hat you’re referring to. The table is formatted properly, the spacing everywhere matches up.

File: bitnami/external-dns/values.yaml (hunk 2 above)
Comment (morremeyer) [reply bitnami/external-dns/values.yaml:44]: I agree, updated the PR with this change.

File: bitnami/external-dns/templates/_helpers.tpl (hunk 3 above)
Comment (morremeyer) [reply bitnami/external-dns/templates/_helpers.tpl:130]: That was an oversight, thanks for pointing it out. Fixed now.

File: bitnami/external-dns/templates/secret.yaml (hunk 4 above)
Comment (morremeyer) [reply bitnami/external-dns/templates/secret.yaml:29]: Check was added :)

**Comment by morremeyer** [comment]
Thanks for the review @miguelaeh. I updated the PR with the requested changes. I'm unsure what you’re referring to regarding the formatting, so that one is still open.

Can you clarify that?

File: bitnami/external-dns/README.md (hunk 1 above)
Comment (miguelaeh) [reply bitnami/external-dns/README.md:123]: It is just to maintain the table format. Anyway, don't worry, we usually format it when we create a new PR and it is not an issue

**Review by miguelaeh — APPROVED** [review]
LGTM!

File: bitnami/external-dns/README.md (hunk 1 above)
Comment (morremeyer) [reply bitnami/external-dns/README.md:123]: Thanks for merging! I got the table format, but that’s what confused me about your request. The format is correct, it just looks weird in the small preview GitHub shows us here.

In the actual files, it looks like this:

screenshot of the raw file

**Pull request merged**